baseplate/integration/pyramid.py

```diff
@@ -69,9 +77,12 @@ class BaseplateConfigurator(object):
     """
 
     # TODO: remove the default on trust_trace_headers once apps are updated
-    def __init__(self, baseplate, trust_trace_headers=False):
+    def __init__(self, baseplate, trust_trace_headers=False,
+                 auth_factory=None):
         self.baseplate = baseplate
         self.trust_trace_headers = trust_trace_headers
+        self.auth_factory = auth_factory if auth_factory \
+            else AuthenticationContextFactory()
```

Comment:
This is mostly to gracefully handle any downstream services that are not setup to parse and handle authentication tokens from having issues propagating the headers further.  If this conditional is reached, attempting to read the token will cause errors due to undefined secrets in the handler.